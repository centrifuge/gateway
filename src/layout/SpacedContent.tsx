import { Box, Grid } from 'grommet';
import React, { FunctionComponent } from 'react';

const SpacedContent: FunctionComponent = ({ children }) => (
  <Grid
    rows={['flex']}
    columns={['auto', 'xlarge', 'auto']}
    areas={[
      { name: 'left-spacer', start: [0, 0], end: [0, 0] },
      { name: 'content', start: [1, 0], end: [1, 0] },
      { name: 'right-spacer', start: [2, 0], end: [2, 0] },
    ]}
    fill="true"
  >
    <Box gridArea="left-spacer" />
    <Box gridArea="content">{children}</Box>
    <Box gridArea="right-spacer" />
  </Grid>
);

SpacedContent.displayName = 'SpacedContent';

export default SpacedContent;
