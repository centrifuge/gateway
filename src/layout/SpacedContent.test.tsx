import React from 'react';
import { shallow } from 'enzyme';

import SpacedContent from './SpacedContent';

describe('SpacedContent', () => {
  it('matches snapshot', () => {
    const bodyShallow = shallow(<SpacedContent />);
    expect(bodyShallow).toMatchSnapshot();
  });
});
