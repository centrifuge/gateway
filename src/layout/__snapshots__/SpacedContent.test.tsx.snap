// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpacedContent matches snapshot 1`] = `
<Grid
  areas={
    Array [
      Object {
        "end": Array [
          0,
          0,
        ],
        "name": "left-spacer",
        "start": Array [
          0,
          0,
        ],
      },
      Object {
        "end": Array [
          1,
          0,
        ],
        "name": "content",
        "start": Array [
          1,
          0,
        ],
      },
      Object {
        "end": Array [
          2,
          0,
        ],
        "name": "right-spacer",
        "start": Array [
          2,
          0,
        ],
      },
    ]
  }
  columns={
    Array [
      "auto",
      "xlarge",
      "auto",
    ]
  }
  fill="true"
  rows={
    Array [
      "flex",
    ]
  }
>
  <Box
    gridArea="left-spacer"
  />
  <Box
    gridArea="content"
  />
  <Box
    gridArea="right-spacer"
  />
</Grid>
`;
